File: prototype_mod/common/smart_belt/DragDirection.ts
export const enum DragDirection {
  Forward = "forward",
  Backward = "backward",
}

export function directionMultiplier(direction: DragDirection): number {
  return direction === DragDirection.Forward ? 1 : -1
}

export function swapIfBackwards<T>(
  direction: DragDirection,
  a: T,
  b: T,
): [T, T] {
  return direction === DragDirection.Forward ? [a, b] : [b, a]
}

File: prototype_abstract/src/smart_belt/drag_direction.rs
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum RaySense {
    Forward,
    Backward,
}

impl RaySense {
    pub fn direction_multiplier(self) -> i32 {
        match self {
            RaySense::Forward => 1,
            RaySense::Backward => -1,
        }
    }

    pub fn swap_if_backwards<T>(self, a: T, b: T) -> (T, T) {
        match self {
            RaySense::Forward => (a, b),
            RaySense::Backward => (b, a),
        }
    }
}
